ethereum/contracts/bridgehub/bridgehub-deps/BridgehubMailbox.sol

```diff
@@ -0,0 +1,141 @@
+        uint256 _chainId,
+        uint256 _batchNumber,
+        uint256 _index,
+        L2Log memory _log,
+        bytes32[] calldata _proof
+    ) external view override returns (bool) {
+        address stateTransitionChain = bridgehubStorage.stateTransitionChain[_chainId];
+        return IStateTransitionChain(stateTransitionChain).proveL2LogInclusion(_batchNumber, _index, _log, _proof);
+    }
+
+    function proveL1ToL2TransactionStatus(
+        uint256 _chainId,
+        bytes32 _l2TxHash,
+        uint256 _l2BlockNumber,
+        uint256 _l2MessageIndex,
+        uint16 _l2TxNumberInBlock,
+        bytes32[] calldata _merkleProof,
+        TxStatus _status
+    ) external view override returns (bool) {
+        address stateTransitionChain = bridgehubStorage.stateTransitionChain[_chainId];
+        return
+            IStateTransitionChain(stateTransitionChain).proveL1ToL2TransactionStatus(
+                _l2TxHash,
+                _l2BlockNumber,
+                _l2MessageIndex,
+                _l2TxNumberInBlock,
+                _merkleProof,
+                _status
+            );
+    }
+
+    function l2TransactionBaseCost(
+        uint256 _chainId,
+        uint256 _gasPrice,
+        uint256 _l2GasLimit,
+        uint256 _l2GasPerPubdataByteLimit
+    ) external view returns (uint256) {
+        require(address(1) != address(0), "zero addres");
+
+        address stateTransitionChain = bridgehubStorage.stateTransitionChain[_chainId];
+        return
+            IStateTransitionChain(stateTransitionChain).l2TransactionBaseCost(
+                _gasPrice,
+                _l2GasLimit,
+                _l2GasPerPubdataByteLimit
+            );
+    }
+
+    function requestL2Transaction(
+        uint256 _chainId,
+        address _contractL2,
+        uint256 _l2Value,
+        bytes calldata _calldata,
+        uint256 _l2GasLimit,
+        uint256 _l2GasPerPubdataByteLimit,
+        bytes[] calldata _factoryDeps,
+        address _refundRecipient
+    ) public payable override returns (bytes32 canonicalTxHash) {
+        address stateTransitionChain = bridgehubStorage.stateTransitionChain[_chainId];
+        canonicalTxHash = IStateTransitionChain(stateTransitionChain).requestL2TransactionBridgehub(
+            msg.value,
+            msg.sender,
+            _contractL2,
+            _l2Value,
+            _calldata,
+            _l2GasLimit,
+            _l2GasPerPubdataByteLimit,
+            _factoryDeps,
+            _refundRecipient
+        );
+    }
+
+    function finalizeEthWithdrawal(
+        uint256 _chainId,
+        uint256 _l2BlockNumber,
+        uint256 _l2MessageIndex,
+        uint16 _l2TxNumberInBlock,
+        bytes calldata _message,
+        bytes32[] calldata _merkleProof
+    ) external override {
+        address stateTransitionChain = bridgehubStorage.stateTransitionChain[_chainId];
+        return
+            IStateTransitionChain(stateTransitionChain).finalizeEthWithdrawalBridgehub(
```

Comment:
Imagine the following scenario:

- There is a `StateTransitionChain` that we don't manually control (for instance, we are not its governor). 
- This chain becomes malicious (maybe due to a bug that allows infinite money printing). 
- There are 1000s of stateTransitionChains from that depend on this state transition

What do we do? Right now, we would have to turn off each `_chainId` separately.

I think that we should add a requirement that the `StateTransitionChain` is still trusted for any Mailbox/Bridge related methods

Reply:
We will control all STC's until ZK IP, as we will control all STs, and only STs can launch STCs. 

An contract cannot just register itself in the Bridgehub, or in the ST.

Reply:
@kelemeno I understand about STs, but we might potentially have hundreds of STCs & disabling all of those might not fit into an Ethereum block. We can have a temporary solution, where we iteratively turn off outdated STCs one-by-one, but I think in this particular a more long-term solution would be preferred, since otherwise we would need to constantly worry to not add more than an a certain number of STCs

Reply:
added.